src/sources/dnstap/parser.rs

```diff
@@ -1075,6 +1104,37 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_parse_dnstap_data_with_ede_options() {
+        let mut log_event = LogEvent::default();
+        let raw_dnstap_data = "ChVqYW1lcy1WaXJ0dWFsLU1hY2hpbmUSC0JJTkQgOS4xNi4zGgBy5wEIAxACGAEiEAAAAAAAAAAAAAAAAAAAAAAqECABBQJwlAAAAAAAAAAAADAw8+0CODVA7+zq9wVNMU3WNlI2kwIAAAABAAAAAAABCWZhY2Vib29rMQNjb20AAAEAAQAAKQIAAACAAAAMAAoACOxjCAG9zVgzWgUDY29tAGAAbQAAAAByZLM4AAAAAQAAAAAAAQJoNQdleGFtcGxlA2NvbQAABgABAAApBNABAUAAADkADwA1AAlubyBTRVAgbWF0Y2hpbmcgdGhlIERTIGZvdW5kIGZvciBkbnNzZWMtZmFpbGVkLm9yZy54AQ==";
```

Comment:
How was this test data produced?

Reply:
For this one I haven't been able to reproduce some EDE errors and capture the DNSTAP output, so I have picked up one of the examples above, decoded it and changed response data to contain an EDE error. In this case specifically it was the following response data:
```
id 45880
opcode QUERY
rcode NOERROR
flags
;QUESTION
h5.example.com. IN SOA
;ANSWER
;AUTHORITY
;ADDITIONAL
. 16859136 CLASS1232 TYPE41 \# 57 000f003500096E6F20534550206D61746368696E672074686520445320666F756E6420666F7220646E737365632D6661696C65642E6F72672e
```